src/main/java/com/amazon/opendistroforelasticsearch/sql/esdomain/mapping/FieldMapping.java

```diff
@@ -0,0 +1,134 @@
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.esdomain.mapping;
+
+import com.amazon.opendistroforelasticsearch.sql.domain.Field;
+import com.amazon.opendistroforelasticsearch.sql.utils.StringUtils;
+
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+import static org.elasticsearch.action.admin.indices.mapping.get.GetFieldMappingsResponse.FieldMappingMetaData;
+
+/**
+ * Field mapping that parse native ES mapping.
+ *
+ * NOTE that approaches in this class are NOT reliable because of the ES mapping query API used.
+ * We should deprecate this in future and parse field mapping in more solid way.
+ */
+public class FieldMapping {
+
+    /** Field (name) to be parsed */
+    private final String fieldName;
+
+    /** Native mapping information returned from ES */
+    private final Map<String, FieldMappingMetaData> typeMappings;
+
+    /** Map of field name to Field object that specified in query explicitly */
+    private final Map<String, Field> specifiedFieldByNames;
+
+
+    public FieldMapping(String fieldName) {
+        this(fieldName, emptyMap(), emptyMap());
+    }
+
+    public FieldMapping(String fieldName,
+                        Map<String, FieldMappingMetaData> typeMappings,
+                        Map<String, Field> specifiedFieldByNames) {
+
+        this.fieldName = fieldName;
+        this.typeMappings = typeMappings;
+        this.specifiedFieldByNames = specifiedFieldByNames;
+    }
+
+    /**
+     * Is field specified explicitly in query
+     * @return true if specified
+     */
+    public boolean isSpecified() {
+        return specifiedFieldByNames.containsKey(fieldName);
+    }
+
+    /**
+     * Is field a property field, which means either object field or nested field.
+     * @return true for property field
+     */
+    public boolean isPropertyField() {
+        int numOfDots = StringUtils.countMatches(fieldName, '.');
+        return numOfDots > 1 || (numOfDots == 1 && !isMultiField());
+    }
+
+    /**
+     * Path of property field, for example "employee" in "employee.manager"
+     * @return path of property field
+     */
+    public String path() {
+        int lastDot = fieldName.lastIndexOf(".");
+        return fieldName.substring(0, lastDot);
+    }
+
+    /**
+     * Verify if property field matches wildcard pattern specified in query
+     * @return true if matched
+     */
+    public boolean isWildcardSpecified() {
+        return specifiedFieldByNames.containsKey(path() + ".*");
+    }
+
+    /**
+     * Is field a/in multi-field, for example, field "a.keyword" in field "a"
+     * @return true for multi field
+     */
+    public boolean isMultiField() {
+        return fieldName.endsWith(".keyword");
```

Comment:
I am thinking if there is a better way to identify a multi-field. Because the name of the extra field can be anything else, not just `keyword`. So either let's rename this method if we only care about `.keyword` case, or find a better way.

Reply:
> I am thinking if there is a better way to identify a multi-field. Because the name of the extra field can be anything else, not just `keyword`. So either let's rename this method if we only care about `.keyword` case, or find a better way.

I'd like to keep this as it is because this is the intention of the API (although our approach is bad). For now, it's hard to figure this out because the mapping returned from ES is "flat". I'm thinking of integrating this domain with the existing one `FieldMappings`(https://github.com/opendistro-for-elasticsearch/sql/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/sql/esdomain/LocalClusterState.java#L392) which depends on different format of ES mapping.

Reply:
sounds good